internal/component/loki/process/process.go

```diff
@@ -138,8 +137,9 @@ func (c *Component) Update(args component.Arguments) error {
 		if err != nil {
 			return err
 		}
-		c.entryHandler = loki.NewEntryHandler(c.processOut, func() { pipeline.Cleanup() })
-		c.processIn = pipeline.Wrap(c.entryHandler).Chan()
+		entryHandler := loki.NewEntryHandler(c.processOut, func() { pipeline.Cleanup() })
+		c.entryHandler = pipeline.Wrap(entryHandler)
```

Comment:
I see that pipeline.Wrap starts goroutines and it seems like we were not calling `Stop()` on the wrapped pipeline before. LMK if I'm missing something here.

Reply:
Yes, we didn't call `Stop()` for the new entry handler (the one coming from `Wrap()`) but rather on the first entry handler. For everything to close properly, I think the "outmost" entry handler must be closed.